copy/copy.go

```diff
@@ -546,6 +551,25 @@ func (c *copier) copyOneImage(ctx context.Context, policyContext *signature.Poli
 		return nil, "", "", errors.Wrapf(err, "Error initializing image from source %s", transports.ImageName(c.rawSource.Reference()))
 	}
 
+	options.DestinationManifestCheck = true // TODO: Forced enablement for initial PR testing
+	if options.DestinationManifestCheck && options.SignBy == "" {
```

Comment:
The full set of conditions AFAICS:
- Caller opts in
- No need to edit the image/manifest (per the `reflect.DeepEqual` check in `copyUpdatedConfigAndManifest`, which is non-trivial enough that it should probably be split into a separate `ic.noPendingManifestUpdates` helper instead of copy&pasted):
  - No manifest conversion is needed
  - No need to update the embedded Docker reference
- No need to copy existing signatures (`len(sigs) == 0`, or alternatively all of these signatures are known to exist at the destination as well — that can be left to a future PR)
- No new signature (`opts.SignBy == ""`)
- No OCI encryption/decryption (`copier.oci*Config`)

I’m a bit torn on whether to have the conditions inlined here or in a helper function; I'm leaning towards inlining, to make the logic easier to see and to make it more difficult to overlook that this optimization exists.

The “destination already contains the desired manifest” check, OTOH, is conceptually trivial and would be better split into a helper function, to make `copyOneImage` a little less gargantuan.